src/main/java/me/arminb/sara/dao/QuestionDAOImpl.java

```diff
@@ -0,0 +1,181 @@
+package me.arminb.sara.dao;
+
+import com.mongodb.*;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.FindOneAndUpdateOptions;
+import com.mongodb.client.model.UpdateOptions;
+import com.mongodb.client.result.DeleteResult;
+import com.mongodb.client.result.UpdateResult;
+import com.sun.javadoc.Doc;
+import me.arminb.sara.entities.Answer;
+import me.arminb.sara.entities.Question;
+import org.bson.Document;
+import org.bson.types.ObjectId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+import static me.arminb.sara.Constants.*;
+
+@Repository("questionDAO")
+public class QuestionDAOImpl implements QuestionDAO {
+
+    Logger logger = LoggerFactory.getLogger(QuestionDAOImpl.class);
+
+    @Autowired
+    private MongoDatabase database;
+
+
+    private Question mapQuestion(Document questionDoc){
+
+        Question question = new Question();
+
+        DateFormat format = new SimpleDateFormat("EEE MMM d HH:mm:ss z yyyy");
+        Date createDate = null;
+        Date modifiedDate = null;
+        try {
+            if (questionDoc.get("created_date") != null){
```

Comment:
Question doc should not have a created_date field. It should be taken from the doc's object id